### DISCUSSION TITLE
Signal Bot Framework

### DISCUSSION THREAD

**Discussion by filipre** [bbernhard/signal-cli-rest-api#164] [Show and tell]
Hey!

I created a small Python framework on top of the REST API that makes it more convenient to write new bot commands for Signal. You can check it out here: https://github.com/filipre/signalbot. It allows you to react on messages and also to schedule tasks to send messages automatically. 

Internally, it uses the JSON-RPC Beta version which runs very fast on a Raspberry Pi 4. The REST API made it quite convenient to receive and send messages. Again, great job for writing the wrapper! 

Here is an example conversation using the bot:

![photo_2021-10-09 13 40 18](https://user-images.githubusercontent.com/8502198/136656526-1a37a152-5827-4b7f-b448-4a3bd9f3243e.jpeg)

Maybe it helps someone who is also developing Signal bots.

BTW, I'd love to try out the `reaction` branch (https://github.com/bbernhard/signal-cli-rest-api/commit/52b8cdf3f0d15e8d263268320a614a17df0bade5)  since it seems to be implemented already? It would reduce a bit of clutter during some of my commands.

**Comment by bbernhard** [comment]
Very nice - thanks for sharing!

I actually wanted to add a similar functionality a while ago, but never found the time to do it. The idea was to create a generic hooking mechanism that allows to run pre/post command hooking scripts. That would have allowed users to write lua/python/javascript scripts that will be execute before/after a command runs. 

But as I've never found the time to start working on that, I am glad to see your project step up.

In case you want your project mentioned in the [README](https://github.com/bbernhard/signal-cli-rest-api#clients--libraries), feel free to create a PR.

regarding the `reaction` branch: Unfortunately, the changes in the branch do not work apply anymore on the `jsonrpc` branch. But I'll have a look in the next days to make it work again.
